### DISCUSSION TITLE
Roadmap?

### DISCUSSION THREAD

**Discussion by patriciomacadden** [excid3/revise_auth#70] [General]
I found this project to be quite interesting as it's just plain rails so I started contributing a bit. Is there a roadmap or anything? Features we want to add, like, i don't know, replacing other devise features such as locakable, etc? Or how we really want the extensions to be? Or what exactly we will consider an extension vs a config?

Cheers

**Comment by excid3** [comment]
I'm open to ideas. 👍 

I would like to eventually support most of the Devise features. OAuth would be another nice thing to support.

**Comment by patriciomacadden** [comment]
I have a couple of ideas,

Here's a checklist against devise:

- [x] Database authenticatable: hashes and stores a password in the database to validate the authenticity of a user while signing in. The authentication can be done both through POST requests or HTTP Basic Authentication.
- [ ] Omniauthable: adds OmniAuth support. It's not much apart from config and bunch of helpers. Whenever I use this module I end up copying a controller I already have for supporting multiple authentications against a user. So you could register/connect your account in one go or create an account and then connect. It can get tricky if you have extra required fields but we can figure out what to do. It would be nice to add support for this out of the box, though.
- [ ] Confirmable: sends emails with confirmation instructions and verifies whether an account is already confirmed during sign in.
- [x] Recoverable: resets the user password and sends reset instructions.
- [x] Registerable: handles signing up users through a registration process, also allowing them to edit and destroy their account.
- [ ] Rememberable: manages generating and clearing a token for remembering the user from a saved cookie.
- [ ] Trackable: tracks sign in count, timestamps and IP address.
- [ ] Timeoutable: expires sessions that have not been active in a specified period of time.
- [x] Validatable: provides validations of email and password. It's optional and can be customized, so you're able to define your own validations.
- [ ] Lockable: locks an account after a specified number of failed sign-in attempts. Can unlock via email or after a specified time period.

It'd be nice to make this things optional so we don't draw routes for them, for example. Devise, IIRC, does this through the `devise` class method in the model. Should we want to do the same or through the initializer? I don't know.
